### PR TITLE
Extended command line options.

### PR DESCRIPTION
**Pull request by CarlosNihelton**
With PR #123 , #125 and #129 we added a new command line option to the launcher: `--enable-installer`.

This PR presents a more flexible way to add new command line options while preserving the existing ones, including the upstream. It is designed to avoid interaction with the upstream "command line parsing" (a series of if-else clauses in `DistroLauncher.cpp`) by removing the "extended" options from the vector of arguments (which is passed by a mutating reference).

The parsing output is a variant of types representing the tasks the launcher is supposed to do plus a `std::monostate` to represent the not-our-business options, which must be handled by the upstream code. Think of `std::monostate` like a `nil` type for variants.

Shall we want to support more command lines arguments in the future, it's just a matter of creating one new subtype in this file  and a tryParse function mimicking the existing implementations.

You'll notice a contrieved usage of templates inside the cpp file. That doesn't leak to the interface. It's used because for most of the options we added at this time are parsed in the very same way, except for `--autoinstall`, varying only in the quantity and contents of the required strings.

There are better command line parsing libraries out there, like `boost::program_options` or [docopt.cpp](https://github.com/docopt/docopt.cpp), but we are still avoiding bringing dependencies to the launcher and also, this requirement on not disturbing the "upstream CLI parsing" would require some customization anyway.

This, as well as the `MSG_USAGE_EXTENDED` message will be brought together by my next PR.

### CODE REVIEW

**Comment by CarlosNihelton** [comment]
Besides the tests contained in this PR, the intended usage of this command line parsing mechanism can be seen [in this branch](https://github.com/ubuntu/WSL/blob/deeng-179-application/DistroLauncher/Application.h).

You can also see the final interconnection [in the same branch](https://github.com/ubuntu/WSL/blob/deeng-179-application/DistroLauncher/DistroLauncher.cpp).

File: DistroLauncher/extended_cli_parser.cpp
Comment (didrocks) [inline DistroLauncher/extended_cli_parser.cpp:2]: 2022, should be done in a future update :)

File: DistroLauncher/messages.mc
Comment (didrocks) [inline DistroLauncher/messages.mc:113]: I think this should rather be something like `--gui` or `--oobe` as the command is `ubuntu.exe install <with something>`. Also, we should consider forcing or not TUI.

We should review all together the UX for this, but let’s not block on it for now.

**Review by jibel — APPROVED** [review]
Some thoughts for future enhancements in comments. Otherwise LGTM. Thanks!

**Review by didrocks — APPROVED** [review]

**Pull request merged**
